### PR TITLE
docs: Update node config comments so that they can be used for doc generation.

### PR DESCRIPTION
**Pull request by winder**
## Summary

Updating comments on `config.Local` so that they can be used to generate the [node configuration page](https://developer.algorand.org/docs/run-a-node/reference/config/#algod-configuration-settings). Here I've mainly chosen the best content from the developer portal and the existing comments.

Followup PRs should be made:
* Better and more consistent formatting. Especially in cases where multiple options are listed.
* Remove unused settings. I noticed at least 5 which don't seem to be used. I only marked the ones which had minimal or missing documentation.
* Generally going through the entire list to clarify and update documentation as needed.

## Test Plan

Manually ran the [config_json_gen](https://github.com/algorand/docs/tree/staging/scripts/config_json_gen) and verified that the generated markdown document looks good and has no warnings.

### CODE REVIEW

File: config/localTemplate.go [hunk 1]
```diff
@@ -149,29 +153,35 @@ type Local struct {
...
+	// RestWriteTimeoutSeconds is passed to the API servers rest http.Server implementation.
 	RestWriteTimeoutSeconds int `version[4]:"120"`
 
-	// SRV-based phonebook
+	// DNSBootstrapID specifies the names of a set of DNS SRV records that identify the set of nodes available to connect to.
+	// When resolving the bootstrap ID <network> will be replaced by the genesis block's network name. This string uses a URL
+	// parsing library and supports optional backup and dedup parameters. 'backup' is used to provide a second DNS entry to use
+	// in case the primary is unavailable. dedup is intended to be used to deduplicate SRV records returned from the primary
+	// and backup DNS address. If the <name> macro is used in the dedup mask, it must be at the beginning of the expression.
+	// This is not typically something a user would configure. For more information see config/dnsbootstrap.go.
```
Comment (winder) [inline config/localTemplate.go:178]: @algogm DNSBootstrapID was one of the mis-documented settings. I copied over some of the highlights from dnsbootstrap.go, how does this look?

File: config/localTemplate.go [hunk 2]
```diff
@@ -44,39 +44,43 @@ type Local struct {
...
 
-	// environmental (may be overridden)
-	// When enabled, stores blocks indefinitely, otherwise, only the most recent blocks
-	// are being kept around. ( the precise number of recent blocks depends on the consensus parameters )
+	// Archival nodes retain a full copy of the block history. Non-Archival nodes will delete old blocks and only retain what's need to properly validate blockchain messages (the precise number of recent blocks depends on the consensus parameters. Currently the last 1321 blocks are required). This means that non-Archival nodes require significantly less storage than Archival nodes. Relays (nodes with a valid NetAddress) are always Archival, regardless of this setting. This may change in the future. If setting this to true for the first time, the existing ledger may need to be deleted to get the historical values stored as the setting only effects current blocks forward. To do this, shutdown the node and delete all .sqlite files within the data/testnet-version directory, except the crash.sqlite file. Restart the node and wait for the node to sync.
```
Comment (algorandskiy) [inline config/localTemplate.go:47]: maybe a multi-line comment instead?

**Review by algorandskiy — APPROVED** [review]

File: config/localTemplate.go (hunk 2 above)
Comment (winder) [reply config/localTemplate.go:47]: This should be a future PR. I think the formatting for a lot of them could be improved, for this PR I'm trying to get the bare minimum in place. The code generator probably needs some improvement too as the doc page currently removes all the newlines.

**Review by jannotti — APPROVED** [review]

File: config/localTemplate.go [hunk 3]
```diff
@@ -44,39 +44,43 @@ type Local struct {
...
-	// The public address to connect to that is advertised to other nodes.
+	// PublicAddress is the public address to connect to that is advertised to other nodes.
 	// For MainNet relays, make sure this entry includes the full SRV host name
 	// plus the publicly-accessible port number.
 	// A valid entry will avoid "self-gossip" and is used for identity exchange
-	// to deduplicate redundant connections
+	// to de-duplicate redundant connections
 	PublicAddress string `version[0]:""`
 
+	// MaxConnectionsPerIP is the maximum number of connections allowed per IP address.
```
[the remark this answers is not in the document]
Comment (winder) [reply config/localTemplate.go:66]: Couldn't tell you. This comment was copied from https://developer.algorand.org/docs/run-a-node/reference/config/

**Pull request merged**